src/gwt/src/org/rstudio/studio/client/rsconnect/ui/RSConnectPublishButton.java

```diff
@@ -840,6 +844,46 @@ public void onError(ServerError error)
          }
       });
    }
+
+   // fetch render output info if necessary and then fire deployment event
+   private void fireDeployDocEvent(final String target,
+                                   final RenderedDocPreview docPreview,
+                                   final RSConnectDeploymentRecord previous)
+   {
+      // prevent re-entrancy
+      if (rmdInfoPending_)
+         return;
+      
+      if (StringUtil.isNullOrEmpty(docPreview.getOutputFile()))
+      {
+         rmdInfoPending_ = true;
+         rmdServer_.getRmdOutputInfo(target,
+               new ServerRequestCallback<RmdOutputInfo>()
+               {
+                  @Override
+                  public void onResponseReceived(RmdOutputInfo response)
+                  {
+                     RenderedDocPreview preview = new RenderedDocPreview( target,
+                           response.output_file_exists ? response.output_file : "", true);
+                     events_.fireEvent(RSConnectActionEvent.DeployDocEvent(
+                           preview, RSConnect.CONTENT_TYPE_DOCUMENT, previous));
+                     rmdInfoPending_ = false;
+                  }
+            
+                  @Override
+                  public void onError(ServerError error)
+                  {
+                     Debug.logError(error);
```

Comment:
Should we show this error to the user? (seems like we should since otherwise it will look like nothing is happening when they deploy)